specification/trace/semantic_conventions/instrumentation/elasticsearch.md

```diff
@@ -0,0 +1,50 @@
+# Semantic conventions for Elasticsearch
+
+**Status**: [Experimental](../../../document-status.md)
+
+This document defines semantic conventions to apply when creating a span for requests to Elasticsearch.
+
+## Span Name
+
+The **span name** SHOULD be of the format `<endpoint id>`.
+
+The elasticsearch endpoint identifier is used instead of the url path in order to reduce the cardinality of the span 
+name, as the path could contain dynamic values. The endpoint id is the `name` field in the
+[elasticsearch schema](https://raw.githubusercontent.com/elastic/elasticsearch-specification/main/output/schema/schema.json).
+If the endpoint id is not available, the span name SHOULD be `http.request.method`.
+
+## URL path parts
+
+Many Elasticsearch url paths allow dynamic values. These SHOULD be recorded in span attributes in the format
+`db.elasticsearch.path_parts.<key>`, where `<key>` is the url path part name. The implementation SHOULD
+reference the [elasticsearch schema](https://raw.githubusercontent.com/elastic/elasticsearch-specification/main/output/schema/schema.json)
+in order to map the path part values to their names.
+
+| Attribute                           | Type | Description                           | Examples                                                                                 | Requirement Level |
+|-------------------------------------|---|---------------------------------------|------------------------------------------------------------------------------------------|---|
+| `db.elasticsearch.path_parts.<key>` | string | A dynamic value in the url path.      | `db.elasticsearch.path_parts.index=test-index`; `db.elasticsearch.path_parts.doc_id=123` | Conditionally Required: [1] |
```

Comment:
The `key` here is one of the allowed/known "parameters that can be passed to a given ES URL? Are those an unbounded list or do we know all of them? I'm not sure how I feel about this, since it's not listed in YAML but just defined here. 

One clear benefit for ES is if it adds new parameters, then the conventions here don't need to change. 
But in counterpart, if there aren't many of them would it be better to list them all here and have them part of the YAML definitions? 

Instrumentation usually have constants for such attribute key names and if we use this, then they need to construct such attribute keys "all the time", no? 

Also, the values that are recorded here, since they come from the URL, could they contain sensitive info/invalid data that needs to be sanitized? Do you want to always have this recorded or maybe should this be Opt-in?

Reply:
Hi @joaopgrassi there is indeed a finite list of the url path parameters. You can see them as the elements in `{ }` in the `path` values of the [spec](https://raw.githubusercontent.com/elastic/elasticsearch-specification/main/output/schema/schema.json). They are not runtime dynamic; any path parameters for a given version of ES would be defined in the spec. I don't think it would be a good idea to enumerate them in YAML because 1) there are too many and 2) that would require semantic convention updates with each new release of ES.

The path parameters names themselves would not contain sensitive info, as they only represent keys. The values, however, are either identifiers produced by ES itself or very technical identifiers (e.g. snapshot repositories, template names, etc). The only path value that could potentially contain PII is the document id, as sometimes people put an email address or name there but that's generally bad practice and advised against.

Reply:
Thanks for the info. The problem about code generation of attribute constants is still a problem though. I don't think there's any language using our build-tools to generate the code files containing the attributes from YAML, but at some point I think they will. With the introduction of this attribute that will not be possible (it will be will render an attribute named `db.elasticsearch.path_parts.<key>`). 

I don't know how to go about this now. Either allow this as is and open a separated issue to discuss it, or go with this PR without it and open a separated issue/PR to discuss.

Reply:
I took this approach from [http headers](https://github.com/open-telemetry/semantic-conventions/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers)
Is there a plan for that? I would think we should open a separate issue since this and the http headers approach should be aligned.

Reply:
Just to clarify: I think the approach for generating code files from the `part_parts` notation (and http headers notation) shouldn't block the approval and merging of this PR. What do you think?

Reply:
Ah my bad, I somehow thought it was part of the `yaml` file but I see now it's just in the markdown. I guess then all good, instrumentations will have to do their own thing to record these, and since we have precedence all should be good. Thanks for looking into it!

Reply:
ok, great, thank you for your review!

Reply:
What are the next steps for getting the PR merged?